.github/workflows/release.yaml

```
@@ -1,39 +1,114 @@
 
 jobs:
-  build:
-    name: tag release
+  push_release_tags:
     runs-on: ubuntu-latest
-
-    permissions:
-      contents: write
-
-    if: github.repository == 'metal3-io/ip-address-manager'
+    outputs:
+      release_tag: ${{ steps.release-version.outputs.release_version }}
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@692973e3d937129bcbf40652eb9f2f61becf3332 # tag=v4.1.7
+        with:
+          fetch-depth: 0
+      - name: Get changed files
+        id: changed-files
+        uses: tj-actions/changed-files@e9772d140489982e0e3704fea5ee93d536f1e275 # tag=v45.0.1
+      - name: Get release version
+        id: release-version
+        run: |
+            if [[ ${{ steps.changed-files.outputs.all_changed_files_count }} != 1 ]]; then
+              echo "1 release notes file should be changed to create a release tag, found ${{ steps.changed-files.outputs.all_changed_files_count }}"
+              exit 1
+            fi
+            for changed_file in ${{ steps.changed-files.outputs.all_changed_files }}; do
+              export RELEASE_VERSION=$(echo "${changed_file}" | grep -oP '(?<=/)[^/]+(?=\.md)')
+              echo "RELEASE_VERSION=${RELEASE_VERSION}" >> ${GITHUB_ENV}
+              echo "RELEASE_VERSION=${RELEASE_VERSION}" >> ${GITHUB_OUTPUT}
+              if [[ "${RELEASE_VERSION}" =~ ^v[0-9]+\.[0-9]+\.[0-9]+(-[0-9A-Za-z-]+(\.[0-9A-Za-z-]+)*)?(\+[0-9A-Za-z-]+(\.[0-9A-Za-z-]+)*)?$ ]]; then
+                echo "Valid semver: ${RELEASE_VERSION}"
+              else
+                echo "Invalid semver: ${RELEASE_VERSION}"
+                exit 1
+              fi
+            done
+      - name: Determine the release branch to use
+        run: |
+            if [[ ${RELEASE_VERSION} =~ beta ]] || [[ ${RELEASE_VERSION} =~ alpha ]]; then
+              export RELEASE_BRANCH=main
+              echo "RELEASE_BRANCH=${RELEASE_BRANCH}" >> ${GITHUB_ENV}
+              echo "This is a beta or alpha release, will use release branch ${RELEASE_BRANCH}"
+            else
+              export RELEASE_BRANCH=release-$(echo ${RELEASE_VERSION} | sed -E 's/^v([0-9]+)\.([0-9]+)\..*$/\1.\2/')
+              echo "RELEASE_BRANCH=${RELEASE_BRANCH}" >> ${GITHUB_ENV}
+              echo "This is not a beta or alpha release, will use release branch ${RELEASE_BRANCH}"
+            fi
+      - name: Create or checkout release branch
+        run: |
+          if git show-ref --verify --quiet "refs/remotes/origin/${RELEASE_BRANCH}"; then
+            echo "Branch ${RELEASE_BRANCH} already exists"
+            git checkout "${RELEASE_BRANCH}"
+          else
+            git checkout -b "${RELEASE_BRANCH}"
+            git push origin "${RELEASE_BRANCH}"
+            echo "Created branch ${RELEASE_BRANCH}"
+          fi
+      - name: Validate tag does not already exist
+        run: |
+          if [[ $(git tag -l ${RELEASE_VERSION}) ]]; then
```

Let's make it clear what it tries to achieve.

```suggestion
          if [[ -n "$(git tag -l "${RELEASE_VERSION}")" ]]; then
```